CHANGELOG.md

```diff
@@ -15,8 +15,17 @@ full changeset diff at the end of each section.
 Current Trunk
 -------------
 
+v120
+----
+
+ - Remove closed world validation checks. These checks were causing more harm
+   than good. All valid code will now validate with `--closed-world` (but also
+   some invalid code as well; for the long-term plans, see #6965). (#7019)
```

Comment:
What invalid code is this?

Reply:
Code that will break. The type of code we were hoping the validation would warn users about.

Reply:
I don't think we expect any code to break, though, right? It might just be less optimized if more of the types are public. Or is this a warning about buggy optimizations that might crash or misoptimize in the presence of complex public types?

Reply:
The latter: closed world can break code if the code does not actually conform to the claims (i.e. the outside looks into the types of things sent). Just like TrapsNeverHappen can break code if the code traps.

Reply:
I rewrote the comment a bit to make it more explicit.